### PR TITLE
track failed guest installations

### PR DESCRIPTION
**Pull request by tbaev**
- Related ticket: https://progress.opensuse.org/issues/113785
- Needles: N/A
- Verification run:  
[KVM](http://openqa.qam.suse.cz/tests/50045) and [XEN](http://openqa.qam.suse.cz/tests/50043#)
simulated fail runs
[KVM](http://openqa.qam.suse.cz/tests/50049) and [XEN](http://openqa.qam.suse.cz/tests/50051)

### CODE REVIEW

**Comment by tbaev** [comment]
@alice-suse @xguo @waynechen55 @Julie-CAO @RoyCai7 @varun please help to review this PR. Thanks

File: tests/virtualization/universal/waitfor_guests.pm [hunk 1]
```diff
@@ -38,10 +38,23 @@ sub run {
...
+            if (script_run("test -f /tmp/guests_ip/$guest") ne 0) {
+                push(@wait_guests, $guest);
+            }
+        }
+        # if all guests are install exit the loop
+        last if @wait_guests == 0;
+        sleep 120;
+        # if retry number is reached the test will fail
+        if ($count == $retry) {
+            record_info("failed: timeout", "timeout installation for @wait_guests");
```
Comment (alice-suse) [inline tests/virtualization/universal/waitfor_guests.pm:55]: LGTM. Minor comment, better to capitalize the first character in record_info msg.

**Comment by RoyCai7** [comment]
LGTM

File: tests/virtualization/universal/waitfor_guests.pm [hunk 2]
```diff
@@ -38,10 +38,23 @@ sub run {
...
-    script_retry("test -d /tmp/guests_ip", retry => 15, delay => 120);
-    foreach my $guest (@guests) {
-        script_retry("test -f /tmp/guests_ip/$guest", retry => 20, delay => 120);
-        record_info("$guest installed", "Guest installation completed");
+    my $retry = 35;
+    my $count = 0;
+    while ($count++ < $retry) {
+        my @wait_guests = ();
+        foreach my $guest (@guests) {
+            if (script_run("test -f /tmp/guests_ip/$guest") ne 0) {
```
Comment (Julie-CAO) [inline tests/virtualization/universal/waitfor_guests.pm:46]: Actually it is a bit redundent that we check all the guests each time, those passed guest can be skipped. But it does not matter, the codes work anyway, it fits our target. LGTM.

Comment (tbaev) [reply tests/virtualization/universal/waitfor_guests.pm:46]: Good observation Julie, thanks for the comment. I was thinking of making a second array with installed guests and re checking only the missing ones, but in this case checking all guest is not slowing down the test time and code is cleaner/easier to read.

**Pull request merged**